### PR TITLE
Comply IPv6 url with RFC2732.

### PR DESCRIPTION
**Pull request by StevenYCChou**
Literal IPv6 should be enclosed in "[" and "]" in URL. See details in
https://www.ietf.org/rfc/rfc2732.txt.

### CODE REVIEW

File: stackdriver/client_test.go [hunk 1]
```diff
@@ -120,8 +120,20 @@ func TestEmptyRequest(t *testing.T) {
 }
 
 func TestResolver(t *testing.T) {
-	addressesToTest := []string{"stackdriver.invalid", "2001:db8::"}
-	for _, address := range addressesToTest {
+	tests := []struct {
```
Comment (jkohen) [inline stackdriver/client_test.go]: I think this is similar, but the intention is more explicit and it avoids corner cases in the test code below:
	tests := []struct {
		urlHostname       string
		expectedAddress string
	}{
		{
			"stackdriver.invalid",
			"stackdriver.invalid",
		},
		{
			"[2001:db8::]",
			"2001:db8::",
		},
	}

Comment (StevenYCChou) [reply stackdriver/client_test.go]: Updated. I use the term `host` instead of `hostname` to match the definition in RFC2732.

File: stackdriver/client_test.go [hunk 2]
```diff
@@ -164,10 +176,11 @@ func TestResolver(t *testing.T) {
 			t.Fatal(err)
 		}
 		requestedTarget := c.conn.Target()
-		expectedTarget := c.resolver.Scheme() + ":///" + address
+		expectedTarget := fmt.Sprintf("%s:///%s",
+			c.resolver.Scheme(), test.expectedAddress)
 		if requestedTarget != expectedTarget {
-			t.Errorf("ERROR: Remote address is %s, want "+expectedTarget,
-				requestedTarget)
+			t.Errorf("ERROR: get target as %s, want %s",
```
Comment (jkohen) [inline stackdriver/client_test.go]: nit, "get target as"->got; or minimally get->got

Comment (StevenYCChou) [reply stackdriver/client_test.go]: fixed.

**Comment by StevenYCChou** [comment]
It passes all unit tests but fails on COVERALLS_TOKEN, which is unrelated. Will merge this PR.
